src/background/services/balances/BalancesService.ts

```diff
@@ -1,109 +1,103 @@
-    private networkService: NetworkService,
-    private balanceServiceGlacier: BalancesServiceGlacier,
-    private glacierService: GlacierService
+    private settingsService: SettingsService,
+    private moduleManager: ModuleManager
   ) {}
 
-  private getBalanceServiceByProvider(provider: any) {
-    const balanceService = [
-      this.balancesServiceEVM,
-      this.balancesServiceBTC,
-    ].find((service) => {
-      return !!service.getServiceForProvider(provider);
-    });
-
-    if (balanceService) return balanceService;
-    throw new Error('no balances service for this provider is supported');
-  }
-
   async getBalancesForNetwork(
     network: NetworkWithCaipId,
     accounts: Account[],
     priceChanges?: TokensPriceShortData
   ): Promise<Record<string, Record<string, TokenWithBalance>>> {
-    if (isPchainNetwork(network)) {
-      const pChainBalances = await this.balancesServicePVM.getBalances({
-        accounts,
-        network,
-      });
-      return pChainBalances;
-    }
+    const sentryTracker = Sentry.startTransaction(
+      {
+        name: 'BalancesService: getBalances',
+      },
+      {
+        network: network.caipId,
+      }
+    );
+    const module = await this.moduleManager.loadModuleByNetwork(network);
 
-    if (isXchainNetwork(network)) {
-      const xChainBalances = await this.balancesServiceAVM.getBalances({
-        accounts,
-        network,
-      });
-      return xChainBalances;
-    }
+    const currency = (
+      await this.settingsService.getSettings()
+    ).currency.toLowerCase();
 
-    const isSupportedNetwork = await this.glacierService.isNetworkSupported(
-      network.chainId
-    );
+    const rawBalances = await module.getBalances({
+      // TODO: Use public key and module.getAddress instead to make this more modular
+      addresses: accounts
+        .map((account) => {
+          switch (network.vmName) {
+            case NetworkVMType.EVM:
+              return account.addressC;
+            case NetworkVMType.BITCOIN:
+              return account.addressBTC;
+            case NetworkVMType.AVM:
+              return account.addressAVM;
+            case NetworkVMType.PVM:
+              return account.addressPVM;
+            case NetworkVMType.CoreEth:
+              return account.addressCoreEth;
+            default:
+              return undefined;
+          }
+        })
+        .filter((address): address is string => !!address),
+      network,
+      currency,
+    });
 
-    if (isSupportedNetwork) {
-      try {
-        return await this.balanceServiceGlacier.getBalances(
-          accounts,
-          network,
-          priceChanges
-        );
-      } catch (error) {
-        if (error instanceof GlacierUnhealthyError) {
-          this.glacierService.setGlacierToUnhealthy();
-        } else {
-          throw error;
+    // Apply price changes data, VM Modules don't do this yet
+    const balances = Object.keys(rawBalances).reduce(
```

Comment:
Soo, it's possible that we get errors here:

<img width="622" alt="image" src="https://github.com/user-attachments/assets/d0dd9448-7a58-4d81-b4bf-14b154133d00">

Should we handle those?

Reply:
Yes, but not scope of this PR. For now we just ignore the balances with errors and the extension will fallback to using the cache just as before. https://ava-labs.atlassian.net/browse/CP-9224